### DISCUSSION TITLE
Training using TOPAZ train

### DISCUSSION THREAD

**Discussion by Arvi1991** [tbepler/topaz#90] [Q&A]
Hi Alex and Tristan,

Is there a way to continue topaz train from previous failed job, I want to save time that it takes to preprocess micrographs before starting training models.

In addition how do I specify Topaz train/pick/extract to use multiple 2080ti in relion?

thanks
Arvind

**Comment by tbepler** [comment]
Are you asking about the RELION interface specifically?

If not, preprocessing only needs to be done once using the CLI, because you can save the preprocessed micrographs to reuse them. I think cryoSPARC also saves them, but @alexjnoble probably knows better.

Currently, Topaz doesn't support multi-GPU outside of denoising, because I found that train is bottlenecked by generating image transformations on the CPU rather than by the GPU even when only a single GPU is available. It's something I'm interested in revisiting in the future, though, to see if speed improvements are possible.

**Reply by Arvi1991** [reply]
Yes I am asking about relion interface specifically;

There appears to be some confusion; it says numberofparticles for relion_topaz_train and it says number_of_particles for relion_topaz_pick. Is this really true or just a typo;

The relion_topaz_train.py results in preprocessed_micrographs.star. Should this be input for picking job?

**Comment by alexjnoble** [comment]
Hi Arvind,

Yes, there is a simple flag you can set for run_topaz_pick.py called skip_preprocess that should do what you want.

Also, I updated the description for the picking script to remove number_of_particles. Thanks for finally getting me to do this=)

Best,
-Alex
